src/OrchardCore.Modules/OrchardCore.AutoSetup/Extensions/DistributedLockExtensions.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading.Tasks;
+using OrchardCore.Locking;
+using OrchardCore.Locking.Distributed;
+using OrchardCore.AutoSetup.Options;
+
+namespace OrchardCore.AutoSetup.Extensions
+{
+    /// <summary>
+    /// The distributed lock extensions for Auto Setup.
+    /// </summary>
+    public static class DistributedLockExtensions
+    {
+        /// <summary>
+        /// Lock Prefix
+        /// </summary>
+        private const string LockPrefix = "AUTOSETUP_LOCK";
+
+        /// <summary>
+        /// The default timeout in milliseconds to acquire a distributed setup lock.
+        /// </summary>
+        private const int DefaultLockTimeout = 20_000;
+
+        /// <summary>
+        /// The default expiration in milliseconds of the distributed setup lock.
+        /// </summary>
+        private const int DefaultLockExpiration = 20_000;
+
+        /// <summary>
+        /// Tries to acquire a setup lock 
+        /// </summary>
+        /// <param name="distributedLock">
+        /// The distributed Lock.
+        /// </param>
+        /// <param name="shellName">
+        /// The shell Name for which lock acquires.
+        /// </param>
+        /// <param name="lockOptions">
+        /// The setup options.
+        /// </param>
+        /// <returns>
+        /// The <see cref="Task"/>.
+        /// </returns>
+        public static Task<(ILocker locker, bool locked)> TryAcquireAutoSetupLockAsync(this IDistributedLock distributedLock, string shellName, LockOptions lockOptions)
+        {
+            if (lockOptions == null)
+            {
+                throw new ArgumentNullException(nameof(lockOptions));
+            }
+
+            return distributedLock.TryAcquireLockAsync(
+                $"{LockPrefix}_{lockOptions.LockName ?? shellName}",
+                TimeSpan.FromMilliseconds(lockOptions.LockTimeout == 0 ? DefaultLockTimeout : lockOptions.LockTimeout),
+                TimeSpan.FromMilliseconds(lockOptions.LockExpiration == 0 ? DefaultLockExpiration : lockOptions.LockExpiration));
```

Comment:
I would have check `<= 0`, just in case of a negative value

Reply:
So, if at the end we only use the distributed lock in a simple way, maybe we don't need this extension at all.

Unless if we check that it is a `ILocalLock`, as in `TryAcquireShellActivateLockAsync()`, and in that case don't use any timeout and expiration by setting them to `TimeSpan.MaxValue`. I did this for shell because it is a sensitive place, so that if it is a local lock it just works as before with a simple semaphore without any timeout / expiration.

So, the only difference with `TryAcquireShellActivateLockAsync()` would be that we would not have to check if it is `Initializing`, just check if it is a local lock.

But as you want, i'm not against using the distributed lock in the simplest way, but in that case maybe this extension is not necessary.